### PR TITLE
expr/arithmetic: implement div_real && div_decimal

### PR DESCRIPTION
**Pull request by AndreMouche**
@BusyJay @hicqu @XuHuaiyu PTAL

### CODE REVIEW

File: src/coprocessor/dag/expr/arithmetic.rs [hunk 1]
```diff
@@ -144,6 +144,30 @@ impl FnCall {
...
+
+    pub fn divide_decimal<'a, 'b: 'a>(
+        &'b self,
+        ctx: &StatementContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, Decimal>>> {
+        let lhs = try_opt!(self.children[0].eval_decimal(ctx, row));
+        let rhs = try_opt!(self.children[1].eval_decimal(ctx, row));
+        match lhs.into_owned() / rhs.into_owned() {
+            Some(v) => Ok(Some(Cow::Owned(v.unwrap()))),
```
Comment (BusyJay) [inline src/coprocessor/dag/expr/arithmetic.rs]: Need to check truncated or overflow here?

File: src/coprocessor/dag/expr/arithmetic.rs [hunk 2]
```diff
@@ -249,6 +273,19 @@ mod test {
                 Datum::F64(-0.01),
                 Datum::F64(-0.0101001),
             ),
+            // TODO: support precision in divide.
+            // (
+            //     ScalarFuncSig::DivideReal,
```
Comment (BusyJay) [inline src/coprocessor/dag/expr/arithmetic.rs]: I think you can add other cases for real division.

File: src/coprocessor/dag/expr/arithmetic.rs [hunk 3]
```diff
@@ -144,6 +144,30 @@ impl FnCall {
         };
         res.ok_or(Error::Overflow).map(Some)
     }
+
+    pub fn divide_real(&self, ctx: &StatementContext, row: &[Datum]) -> Result<Option<f64>> {
+        let lhs = try_opt!(self.children[0].eval_real(ctx, row));
+        let rhs = try_opt!(self.children[1].eval_real(ctx, row));
+        let res = lhs / rhs;
```
Comment (BusyJay) [inline src/coprocessor/dag/expr/arithmetic.rs]: What if `rhs` is 0?

**Comment by hanfei1991** [comment]
LGTM

**Pull request merged**